src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1509,19 +1523,29 @@ protected void Log(string msg, WarningLevel severity)
 
         public void UpdateWithExtraViewInfo(ExtraViewInfo viewInfo)
         {
+            X = viewInfo.X;
+            Y = viewInfo.Y;
+            Zoom = viewInfo.Zoom; 
+
+            foreach (ExtraNodeViewInfo nodeInfo in viewInfo.NodeViews)
+            {
+                Guid guidValue = Guid.Parse(nodeInfo.Id);
```

Comment:
be careful here, if this value is being read directly from JSON it is only required that the ID is a unique string within the graph, so you may need to use a conversion method to turn this into a guid. See this converter
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/Workspaces/SerializationConverters.cs#L469

I haven't looked hard enough at this code yet to see how you are deserializing the nodeView into this object to know if this used or not already.